dataikuapi/dss/ml.py

```diff
@@ -315,43 +319,951 @@ def save(self):
                 "POST", "/projects/%s/models/lab/%s/%s/settings" % (self.project_key, self.analysis_id, self.mltask_id),
                 body = self.mltask_settings)
 
+
+class HyperparameterSearchSettings(object):
+
+    def __init__(self, raw_settings):
+        self._raw_settings = raw_settings
+
+    def _key_repr(self, key):
+        if isinstance(self._raw_settings[key], string_types):
+            return "    \"{}\"=\"{}\"\n".format(key, self._raw_settings[key])
+        else:
+            return "    \"{}\"={}\n".format(key, self._raw_settings[key])
+
+    def _repr_html_(self):
+
+        res = "<pre>"
+        res += self.__class__.__name__ + "(\n"
+        res += "Search Strategy:\n"
+        res += self._key_repr("strategy")
+        if self._raw_settings["strategy"] == "BAYESIAN":
+            res += self._key_repr("bayesianOptimizer")
+
+        res += "Search Validation:\n"
+        res += self._key_repr("mode")
+        if self._raw_settings["mode"] in {"SHUFFLE", "TIME_SERIES_SINGLE_SPLIT"}:
+            res += self._key_repr("splitRatio")
+        elif self._raw_settings["mode"] in {"KFOLD", "TIME_SERIES_KFOLD"}:
+            res += self._key_repr("nFolds")
+
+        res += self._key_repr("stratified")
+        # TODO: pointers to ssdSeed and time variable ?
```

Comment:
More of an open question. 
They are not direct entries of the underlying `dict` (`_raw_settings`), but accessible in the larger `MLTaskSettings`. Displaying the time variable and the seed used for train/test split and also for validation splits will probably add value to the user (at the cost of keeping a pointer to the parent `MLTaskSettings` in the current `HyperparameterSearchSettings`), but setting them may not be appropriate here. We could do the same as with `strategy` in `PredictionAlgorithmSettings` and have a read-only property with pointer to the relevant setters.

Reply:
IMO, keeping a pointer to the `MLTaskSettings` only to expand the `__repr__` is a bit overkill, especially that these variables are directly accessible through the `MLTaskSettings`.
Do you have a specific use case where the user will only print the Algo settings ?

Reply:
I created a [CH card](https://app.clubhouse.io/dataiku/story/58357/make-the-seed-of-the-hyperparameter-search-independent-from-the-seed-of-the-train-test-split) to address the `seed` issue, which is currently a bit dysfunctional IMO.

Regarding the time ordering variable, I agree that there is not much room for handling it from the `HyperparameterSearchSettings` instead of the dedicated `PredictionSplitParamsHandler`.